packages/inference/src/snippets/python.ts

```diff
@@ -1,480 +1,188 @@
-import { openAIbaseUrl, type SnippetInferenceProvider } from "@huggingface/tasks";
 import type { PipelineType, WidgetType } from "@huggingface/tasks/src/pipelines.js";
 import type { ChatCompletionInputMessage, GenerationParameters } from "@huggingface/tasks/src/tasks/index.js";
 import {
+	openAIbaseUrl,
 	type InferenceSnippet,
 	type ModelDataMinimal,
 	getModelInputSnippet,
 	stringifyGenerationConfig,
 	stringifyMessages,
 } from "@huggingface/tasks";
+import type { InferenceProvider } from "../types";
+import { Template } from "@huggingface/jinja";
+import fs from "fs";
+import path from "path";
+import { existsSync as pathExists } from "node:fs";
+
+const TOOLS = ["huggingface_hub", "requests", "fal_client", "openai"];
+
+type InputPreparationFn = (model: ModelDataMinimal, opts?: Record<string, unknown>) => string | object;
+interface TemplateParams {
+	accessToken?: string;
+	baseUrl?: string;
+	inputs?: string | object;
+	model?: ModelDataMinimal;
+	provider?: InferenceProvider;
+	providerModelId?: string;
+	methodName?: string; // specific to snippetBasic
+	importBase64?: boolean; // specific to snippetImportRequests
+}
+
+// Helpers to find + load templates
+
+const rootDirFinder = (): string => {
+	let currentPath = path.normalize(import.meta.url).replace("file:", "");
```

Comment:
I'm curious if this will work with moon (`import.meta.url`), or if it's finally time to move from CJS to ESM cc @Pierrci

Reply:
are you simply curious or actually wishing for it? x)

Reply:
(I think it will be fine, but I may be wrong!)

Reply:
I think it'll be empty but we'll see!

Reply:
would https://github.com/huggingface/huggingface.js/pull/1255/commits/f6d81d6b33b18f89b81ffda040e55d60b4f6bc43 be ok ?